### PR TITLE
add aloglia search

### PR DESCRIPTION
**Pull request by aabedraba**
I requested a free account from algolia and they accepted it. 
This PR uses algolia now, which is a much better experience

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -140,6 +142,15 @@ const config = {
         theme: lightCodeTheme,
         darkTheme: darkCodeTheme,
       },
+      algolia: {
+        // The application ID provided by Algolia
+        appId: "2K9J967OUZ",
+
+        // Public API key: it is safe to commit it
+        apiKey: "56fe41abaf25479f62d1b2b15aac11b7",
```
Comment (jasmingacic) [inline docs/docusaurus.config.js:150]: Should this be public?

Comment (aabedraba) [reply docs/docusaurus.config.js:150]: yeah, it's okay. I put a comment on top of it.

Comment (mbana) [reply docs/docusaurus.config.js:150]: https://docusaurus.io/docs/search#connecting-algolia

```json
module.exports = {
  // ...
  themeConfig: {
    // ...
    algolia: {
      // The application ID provided by Algolia
      appId: 'YOUR_APP_ID',

      // Public API key: it is safe to commit it
      apiKey: 'YOUR_SEARCH_API_KEY',

      indexName: 'YOUR_INDEX_NAME',

      // Optional: see doc section below
      contextualSearch: true,

      // Optional: Specify domains where the navigation should occur through window.location instead on history.push. Useful when our Algolia config crawls multiple documentation sites and we want to navigate with window.location.href to them.
      externalUrlRegex: 'external\\.com|domain\\.com',

      // Optional: Algolia search parameters
      searchParameters: {},

      // Optional: path for search page that enabled by default (`false` to disable it)
      searchPagePath: 'search',

      //... other Algolia params
    },
  },
};
```

**Review by mbana — APPROVED** [review]
LGTM.

From https://docusaurus.io/docs/search#using-algolia-docsearch:

> DocSearch crawls your website once a week (the schedule is configurable from the web interface) and aggregates all the content in an Algolia index. This content is then queried directly from your front-end using the Algolia API.

Once a week is a long time. Did you configure it to crawl more often?

How do we view the result of this PR, i.e., the change to algolia search.

**Comment by aabedraba** [comment]
@mbana I'll see if we can crawl faster. 

You can also find the preview link in the automatic comment from Netlify in this PR.

**Pull request merged**